### PR TITLE
feat: fix linting errors + add linting workflows

### PR DESCRIPTION
**Pull request by cameronmeissner**
adds desired linting from https://github.com/Azure/aks-tls-bootstrap/issues/9

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -19,73 +19,89 @@ import (
...
 	nextProto  string
+
+	execCredential *datamodel.ExecCredential
+	pemCAs         []byte
+	tlsConfig      *tls.Config
+	azureConfig    *datamodel.KubeletAzureJSON
 }
 
-func (c *tlsBootstrapClientImpl) GetBootstrapToken() (string, error) {
+func (c *tlsBootstrapClientImpl) init() error {
```
Comment (alexeldeib) [inline pkg/client/client.go:52]: this feels like a constructor pretending it has no dependencies but okay with me

File: pkg/client/client.go [hunk 2]
```diff
@@ -19,73 +19,89 @@ import (
...
 	execCredential, err := loadExecCredential(c.logger)
```
Comment (alexeldeib) [inline pkg/client/client.go:53]: e.g. here you already depend on the logger being initialized, but do init of properties in mix and match

File: pkg/client/client.go [hunk 3]
```diff
@@ -153,24 +169,26 @@ func loadExecCredential(logger *logrus.Logger) (*datamodel.ExecCredential, error
...
 	tlsRootStore := x509.NewCertPool()
 	ok := tlsRootStore.AppendCertsFromPEM(pemCAs)
 	if !ok {
 		return nil, fmt.Errorf("failed to load cluster root CA(s)")
 	}
 
 	tlsConfig := &tls.Config{
-		RootCAs:            tlsRootStore,
+		RootCAs:    tlsRootStore,
+		MinVersion: tls.VersionTLS13,
```
Comment (alexeldeib) [inline pkg/client/client.go:190]: I'm not sure we can actually force 1.3? if it works ship it I guess

**Review by alexeldeib — DISMISSED** [review]

File: pkg/client/client.go (hunk 3 above)
Comment (cameronmeissner) [reply pkg/client/client.go:190]: yeah I figured we could revisit this if it doesn't end up working

**Comment by cameronmeissner** [comment]
added todos for comments

**Pull request merged**